src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -282,10 +302,15 @@ public boolean perform(AbstractBuild<?,?> build, Launcher launcher, BuildListene
                 console.println(Messages.CopyArtifact_MissingBuild(expandedProject));
                 return isOptional();  // Fail build unless copy is optional
             }
-            FilePath targetDir = build.getWorkspace(), baseTargetDir = targetDir;
+            FilePath targetDir = workspace, baseTargetDir = targetDir;
             if (targetDir == null || !targetDir.exists()) {
-                console.println(Messages.CopyArtifact_MissingWorkspace()); // (see JENKINS-3330)
-                return isOptional();  // Fail build unless copy is optional
+                if (targetDir != null) {
+                    // TODO: What do we do if the workspace dir does not exist, as in the case of workflow (it seems)? Forcing create for now.
+                    targetDir.mkdirs();
```

Comment:
I added this because I was not getting a workspace in the initial tests i.e. the workspace dir did not exist and the step was exiting early.